lib/iris/experimental/ugrid.py

```diff
@@ -2115,6 +2116,242 @@ def face_node(self):
+        # match the available content in the Mesh object.
+        # That is beyond checking here, but in practice all MeshCoords will be
+        # created by methods of the Mesh itself, which will ensure correctness.
+        if not isinstance(mesh, Mesh):
+            msg = (
+                "'mesh' must be an iris.experimental.ugrid.Mesh, "
+                f"got {mesh}."
+            )
+            raise ValueError(msg)
+        # Handled as a readonly ".mesh" property.
+        # NOTE: currently *not* included in metadata. In future it might be.
+        self._mesh = mesh
+
+        if location not in self.VALID_MESH_LOCATIONS:
+            # Effectively, the possible 'valid' locations are defined by the
+            # support provided in the Mesh api, but it does not declare that
+            # in general terms, so we define our own valid list for those.
+            msg = (
+                f"'location' of {location} is not a valid Mesh location', "
+                f"must be one of {self.VALID_MESH_LOCATIONS}."
+            )
+            raise ValueError(msg)
+        # Held in metadata, readable as self.location, but cannot set it.
+        self._metadata_manager.location = location
+
+        if axis not in Mesh.AXES:
+            # The valid axes are defined by the Mesh class.
+            msg = (
+                f"'axis' of {axis} is not a valid Mesh axis', "
+                f"must be one of {Mesh.AXES}."
+            )
+            raise ValueError(msg)
+        # Held in metadata, readable as self.axis, but cannot set it.
+        self._metadata_manager.axis = axis
+
+    # Define accessors for MeshCoord-specific properties mesh/location/axis.
+    # These are all read-only.
+
+    @property
+    def mesh(self):
+        return self._mesh
+
+    @mesh.setter
+    def mesh(self, value):
+        msg = "Cannot set the 'mesh' of a MeshCoord."
+        raise ValueError(msg)
+
+    @property
+    def location(self):
+        return self._metadata_manager.location
+
+    @location.setter
+    def location(self, value):
+        msg = "Cannot set the 'location' of a MeshCoord."
+        raise ValueError(msg)
+
+    @property
+    def axis(self):
+        return self._metadata_manager.axis
+
+    @axis.setter
+    def axis(self, value):
+        msg = "Cannot set the 'axis' of a MeshCoord."
+        raise ValueError(msg)
+
+    # Provide overrides to mimic the Coord-specific properties that are not
+    # supported by MeshCoord, i.e. "coord_sytem" and "climatological".
+    # These mimic the Coord properties, but always return fixed 'null' values,
+    # and cannot be set.
+
+    @property
+    def coord_system(self):
+        """The coordinate-system of a MeshCoord is always 'None'."""
+        return None
+
+    @coord_system.setter
+    def coord_system(self, value):
+        if value is not None:
+            msg = "Cannot set the coordinate-system of a MeshCoord."
+            raise ValueError(msg)
```

Comment:
This one + 'climatalogical' are different :  They must be writable by the parent constructor.

Reply:
I think I have addressed this now as far as I can.
I have removed setters the mesh/location/axis properties.
But I need them for coord_system / climatological, because the parent constructor writes them.
(if I remove them, I get errors).